flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/deployment/ApplicationObserver.java

```diff
@@ -82,11 +79,14 @@ protected boolean observeFlinkCluster(
 
         var jobStatus = flinkApp.getStatus().getJobStatus();
 
+        var previousJobState = jobStatus.getState();
         boolean jobFound =
                 jobStatusObserver.observe(
-                        jobStatus,
+                        flinkApp.getStatus(),
                         deployedConfig,
                         new ApplicationObserverContext(flinkApp, context, deployedConfig));
+        EventUtils.generateEventsOnJobStatusChanged(
+                kubernetesClient, previousJobState, jobStatus, flinkApp);
```

Comment:
I think this method should be called in the JobStatusObserver directly to avoid having it in multiple places. There you already have previousJobState etc.

Reply:
I have tried that way, but it relies on the resource object, It will have to touch some more class to pass it.

Reply:
Wouldn't it be enough to simply change the jobstatus observer to get the full resource object instead of the status only?

Reply:
Good suggestion, I have pushed the commit, PTAL again